examples/mongodb.yaml

```diff
@@ -0,0 +1,92 @@
+    - apiVersion: v1
+      kind: Secret
+      metadata:
+        name: mongodb-{{ .metadata.name }}-credentials
+        namespace: '{{ .metadata.namespace }}'
+      type: Opaque
+      stringData:
+        password: '{{- kget (print "secret/" .metadata.namespace "/mongodb-" .metadata.name "-credentials") "password" | default (random.AlphaNum 16) -}}'
+        admin_password: '{{- kget (print "secret/" .metadata.namespace "/mongodb-" .metadata.name "-credentials") "admin_password" | default (random.AlphaNum 16) -}}'
+
+    - apiVersion: apps/v1
+      kind: StatefulSet
+      metadata:
+        name: mongodb-{{ .metadata.name }}
+        namespace: '{{ .metadata.namespace }}'
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app.kubernetes.io/name: mongodb
+            app.kubernetes.io/instance: '{{ .metadata.name }}'
+        template:
+          metadata:
+            labels:
+              app.kubernetes.io/name: mongodb
+              app.kubernetes.io/instance: '{{ .metadata.name }}'
+          spec:
+            securityContext:
+              fsGroup: 184
+            containers:
+              - image: docker.io/centos/mongodb-36-centos7:1
+                name: mongodb
+                env:
+                  - name: MONGODB_DATABASE
+                    value: '{{ .spec.database }}'
+                  - name: MONGODB_USER
+                    value: '{{ .spec.user }}'
+                  - name: MONGODB_PASSWORD
+                    valueFrom:
+                      secretKeyRef:
+                        name: mongodb-{{ .metadata.name }}-credentials
+                        key: password
+                  - name: MONGODB_ADMIN_PASSWORD
+                    valueFrom:
+                      secretKeyRef:
+                        name: mongodb-{{ .metadata.name }}-credentials
+                        key: admin_password
+                ports:
+                  - containerPort: 27017
+                resources:
+                  requests:
+                    memory: '{{ .spec.memory | default "64Mi" }}'
+                    cpu: '{{ .spec.cpu | default "50m" }}'
+                  limits:
+                    memory: '{{ .spec.memory | default "300Mi" }}'
+                    cpu: '{{ .spec.cpu | default "100m" }}'
+                volumeMounts:
+                  - mountPath: /var/lib/mongodb/data
+                    name: "mongodata-{{ .metadata.name }}"
+            restartPolicy: Always
+        volumeClaimTemplates:
+        - metadata:
+            name: "mongodata-{{ .metadata.name }}"
+          spec:
+            accessModes:
+            - ReadWriteOnce
+            resources:
+              requests:
+                storage: '{{ .spec.storage.size | default "5Gi" }}'
+
+    - apiVersion: v1
+      kind: Service
+      metadata:
+        name: "mongodb-{{ .metadata.name }}"
+        namespace: '{{ .metadata.namespace }}'
+      spec:
+        ports:
+        - name: "27017"
```

Comment:
name is not required when only one port is specified.  If we are going to use a name, maybe a more descriptive string?

Reply:
I've dropped it